client/keepalive.go

```diff
@@ -44,10 +44,13 @@ func (d *drainingReadCloser) Read(p []byte) (n int, err error) {
 }
 
 func (d *drainingReadCloser) Close() error {
+	defer d.rdr.Close()
+
 	// drain buffer
 	if atomic.LoadUint32(&d.seenEOF) != 1 {
-		//#nosec
-		io.Copy(ioutil.Discard, d.rdr)
+		_, err := io.Copy(ioutil.Discard, d.rdr)
+		return err
```

Comment:
if (err != nil) {return err} ?

Reply:
We don't care about the error, this is after the request is completed

Reply:
I can make the change.

I thought read would fail still. Does it make sense to return an error if io.Copy copies something, since EOF is seen?

Reply:
No this is to drain the remaining bytes on a connection where a server is misbehaving. It's important for TCP keepalive to work and allowing golang to reuse file handles. But we don't care about the error because it's not something you can change.